### ISSUE TITLE
docker.service missing, daemon not running?

### ISSUE THREAD

**Issue by Run1e** [docker/for-linux#406]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior
Docker containers should start successfully

### Actual behavior
Container startup fails, service doesn't seem to be running

### Steps to reproduce the behavior
Hey!

Because of a power outage, docker exited abruptly. Upon attempting to restart my containers I'm getting errors:

Attempting to run hello-world:
```
runie@gecko:~/docker$ docker run hello-world
docker: Error response from daemon: grpc: the connection is unavailable.
ERRO[0000] error waiting for container: context canceled
```
```
runie@gecko:~/docker$ systemctl status docker
Unit docker.service could not be found.
runie@gecko:~/docker$ sudo snap start docker
Started.
runie@gecko:~/docker$ docker run hello-world
docker: Error response from daemon: grpc: the connection is unavailable.
ERRO[0000] error waiting for container: context canceled
runie@gecko:~/docker$ sudo snap services
Service         Startup  Current
docker.dockerd  enabled  active
runie@gecko:~/docker$ systemctl status docker
Unit docker.service could not be found.
```
It seems like the service config file doesn't exist?
The only .service file I found with docker in its name is this:
```
runie@gecko:/etc/systemd/system$ cat snap.docker.dockerd.service
[Unit]
# Auto-generated, DO NOT EDIT
Description=Service for snap application docker.dockerd
Requires=snap-docker-179.mount
Wants=network-online.target
After=snap-docker-179.mount network-online.target
X-Snappy=yes

[Service]
ExecStart=/usr/bin/snap run docker.dockerd
SyslogIdentifier=docker.dockerd
Restart=on-failure
WorkingDirectory=/var/snap/docker/179
TimeoutStopSec=30
Type=simple

[Install]
WantedBy=multi-user.target
```
I've exhausted my knowledge on this, so I hope this is the correct place to post this. Thanks in advance!

**Output of `docker version`:**

```
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   a04f55b
 Built:        Thu Sep 21 20:36:57 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   a04f55b
 Built:        Thu Sep 21 20:36:57 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 11
 Running: 0
 Paused: 0
 Stopped: 11
Images: 15
Server Version: 17.06.2-ce
Storage Driver: aufs
 Root Dir: /var/snap/docker/common/var-lib-docker/aufs
 Backing Filesystem: extfs
 Dirs: 80
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: N/A (expected: 6e23458c129b551d5c9871e5174f6b1b7f6d1170)
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-32-generic
Operating System: Ubuntu Core 16
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.664GiB
Name: gecko
ID: BLH7:LMGB:VNED:FSAW:YK2K:INHT:SHH3:GRLS:43PE:RMUM:QPOG:M5ZW
Docker Root Dir: /var/snap/docker/common/var-lib-docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 16
 Goroutines: 22
 System Time: 2018-08-19T01:30:03.58375371Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

Running on ubuntu:
`Welcome to Ubuntu 18.04.1 LTS (GNU/Linux 4.15.0-32-generic x86_64)`

**Comment by yw662** [comment]
There should be a `docker.service` and `docker.socket` in `/lib/systemd/system`, since you are running ubuntu 18.0.4.1 LTS

but not. You should know the reason. That might because of improper installation, or, as you say, power loss.

I have to say, for ubuntu 18.04, these files looks like this:

`/lib/systemd/system/docker.service`

````
[Unit]
Description=Docker Application Container Engine
Documentation=https://docs.docker.com
After=network-online.target docker.socket firewalld.service
Wants=network-online.target
Requires=docker.socket

[Service]
Type=notify
# the default is not to use systemd for cgroups because the delegate issues still
# exists and systemd currently does not support the cgroup feature set required
# for containers run by docker
ExecStart=/usr/bin/dockerd -H fd://
ExecReload=/bin/kill -s HUP $MAINPID
LimitNOFILE=1048576
# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNPROC=infinity
LimitCORE=infinity
# Uncomment TasksMax if your systemd version supports it.
# Only systemd 226 and above support this version.
TasksMax=infinity
TimeoutStartSec=0
# set delegate yes so that systemd does not reset the cgroups of docker containers
Delegate=yes
# kill only the docker process, not all processes in the cgroup
KillMode=process
# restart the docker process if it exits prematurely
Restart=on-failure
StartLimitBurst=3
StartLimitInterval=60s

[Install]
WantedBy=multi-user.target
````

`/lib/systemd/system/docker.socket`

````
[Unit]
Description=Docker Socket for the API
PartOf=docker.service

[Socket]
ListenStream=/var/run/docker.sock
SocketMode=0660
SocketUser=root
SocketGroup=docker

[Install]
WantedBy=sockets.target
````

These are directly from docker-ce deb package. However, use at you own risk.

**Comment by Run1e** [comment]
Pulling the docker-ce package again fixed my issue. Kind of a shame since afaict there wasn't anything to learn from this event, except that docker seemingly halfway uninstalled itself?

Closing the issue.

**Comment by Run1e** [comment]
@yw662 Thanks for the reply. I saw this page on the docker docs: https://docs.docker.com/config/daemon/#docker-daemon-directory

I tried looking for some of the folders that are listed there, but they didn't exist. Among those were:
`/var/lib/docker`
`/etc/docker`

Seeing they didn't exist made me just attempt to reinstall.

Also, the folder `/lib/systemd/system` did not exist prior to the reinstall I just did. 

Again, thanks for your reply.
